cassandra/4.0/modules/ROOT/pages/cassandra-connector-examples.adoc

```diff
@@ -0,0 +1,575 @@
+%dw 2.0
+output application/java
+---
+{
+ "keyspaceName": payload.keyspaceName,
+ "replicationFactor": payload.replicationFactor,
+ "replicationStrategyClass": payload.replicationStrategyClass
+}
+as Object {
+ class: "org.mule.modules.cassandradb.api.CreateKeyspaceInput"
+}
+----
+
+==== Add the Create Keyspace Operation
+
+Add the *Create keyspace* operation to create the new keyspace for the new table: (*Reviewers: please correct and make this text better*)
+
+. From the *Mule Palette* view, select *CassandDB* and drag the *Create keyspace* operation to the right of the *Transform Message* component.
+. In the *Create keyspace* configuration, click the plus sign (*+*) next to the *Connector configuration* field to add a global element.
+. Configure the fields in the global element:
++
+[%header%autowidth.spread]
+|===
+|Field a|Description
+|Host | `${cassandra.Host}`
+|Port | `${cassandra.Port}`
+|Keyspace | `${cassandra.keySpace}`
+|Username | `${cassandra.userName}`
+|Password | `${cassandra.password}`
+|===
++
+. Click *Test Connection* to confirm that Mule can connect with the target Apache Cassandra instance.
++
+If the connection is successful, click *OK* to save the configuration. Otherwise, review and correct any invalid parameters and retest.
++
+. Save the project.
+. Test the app by sending a REST request to the `test/createKeyspace` path on port `8081`.
+
+=== Create a New Table
+
+This flow creates a new table in the target Apache Cassandra database:
+
+.Flow for creating a new table
+image::cassandra-create-table.png[Create table flow]
+
+==== Configure HTTP Listener
+
+Configure HTTP Listener to initiate a Mule flow when a call is made to the `/test/createTable` path on port `8081`:
+
+. Create a new Mule Project in Studio.
+. In Studio, click *HTTP* and drag the *Listener* operation to the canvas.
+. In the *Listener* configuration, click the plus sign *+* next to the *Connector configuration* field to add a global element.
+. Accept the defaults.
+. In the *Listener* configuration, set the *Path* field value to `/test/createTable`.
+
+=== Add a Transform Message Component
+
+Add a *Transform Component* to set Java variables based on the payload received from *HTTP Listener*: ()*Reviewers, is this description correct*)?
+
+. From the Mule Palette view, drag a *Transform Message* component to the right of *Listener*.
+. In the *Transform Message* configuration, overlay the brackets in the *Output* section with this XML:
++
+[source,dataweave,linenums]
+----
+%dw 2.0
+output application/java
+---
+{
+  "columns": payload.columns,
+  "tableName": payload.tableName,
+  "keyspaceName": payload.keyspaceName
+} as Object {
+  class : "org.mule.modules.cassandradb.api.CreateTableInput"
+}
+----
+
+==== Add the Create Table Operation
+
+Add a *Create Table* operation to create a new table based on a POST request that uses the template created in the first flow: (*Reviewers, please make this text more accurate. I am guessing here.*)
```

Comment:
Not sure what that description is but, we need to add a create table operation in ordeer to create that table on the keyspace that was created before, create table operations takes as parameters a keyspace name, a table name, and a list columns that have a name, a type and a flag if they are primary key or not.  I hope I gave you enough info.